src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -658,6 +658,38 @@ def _visit_max_over(self, node: foast.Call, **kwargs) -> foast.Call:
     def _visit_min_over(self, node: foast.Call, **kwargs) -> foast.Call:
         return self._visit_reduction(node, **kwargs)
 
+    def _visit_astype(self, node: foast.Call, **kwargs) -> foast.Call:
+        casted_obj_type = node.args[0].type
+        dtype_obj_type = node.args[1].type
+        if not node.args[1].id.lower() in fbuiltins.TYPE_BUILTINS.__str__() or not (
+            type_info.is_arithmetic(dtype_obj_type.returns)
+            or type_info.is_logical(dtype_obj_type.returns)
+        ):
```

Comment:
```suggestion
        if not node.args[1].id in fbuiltins.TYPE_BUILTIN_NAMES:
```
or maybe a string conversion is needed like
```suggestion
        if not str(node.args[1].id) in fbuiltins.TYPE_BUILTIN_NAMES:
```